src/toil/common.py

```diff
@@ -901,3 +905,50 @@ def cacheDirName(workflowID):
     :return: Name of the cache directory.
     """
     return 'cache-' + workflowID
+
+
+def getSizeOfDirectory(dirToSizeUp):
+    '''
+    This method will walk through a directory and return the cumulative filesize in bytes of all
+    the files in the directory and its subdirectories.
+
+    :param dirToSizeUp: path to the workflow working directory
+    :return: cumulative size in bytes of all files in the directory
+    :rtype: float
+    '''
+    totalSize = 0
+    # The value from running stat on each linked file is equal. To prevent the same file
+    # from being counted multiple times, we save the inodes of files that have more than one
+    # nlink associated with them.
+    linkedFileInodes = set()
+    for dirPath, dirNames, fileNames in os.walk(dirToSizeUp):
+        folderSize = _getDirSize(dirPath, fileNames, linkedFileInodes)
+        totalSize += folderSize
+    return totalSize
+
+
+def _getDirSize(dirPath, fileNames, linkedFileInodes):
+    '''
+    Returns the size of the directory including all linked files provided their inode was not in
+    linkedFileInodes.
+
+    :param str dirPath: Dir path
+    :param list fileNames: Files in the directory
+    :param set linkedFileInodes: set of inode numbers of files that have multiple links, and
+                                 have already been acccounted for.
+    :return: Size of the directory
```

Comment:
But you know, even I don't document my helper functions in that much detail sometimes. OTOH, it does help to document what the set argument is for.